packages/fxa-settings/src/lib/integrations/integration-factory-flags.ts

```diff
@@ -0,0 +1,131 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { Constants } from '../constants';
+import { ModelDataStore, UrlQueryData } from '../model-data';
+import { IntegrationFlags } from './interfaces/integration-flags';
+
+const DEVICE_PAIRING_SUPPLICANT_PATHNAME_REGEXP = /^\/pair\/supp/;
+
+/**
+ * Extrapolates flags from the state of the current data store. The collective state of these flags are used by
+ * the factory to determine what underlying type of integration to create.
+ *
+ * Note: this logic was ported from fxa-content-server app-start.js.
+ */
+export class DefaultIntegrationFactoryFlags implements IntegrationFlags {
+  protected get pathname() {
+    return this.urlQueryData.pathName;
+  }
+  constructor(private urlQueryData: UrlQueryData) {}
+
+  // TODO: Same method on the Relier flags, can we DRY?
+  isDevicePairingAsAuthority() {
+    return (
+      this._searchParam('redirect_uri') ===
+      Constants.DEVICE_PAIRING_AUTHORITY_REDIRECT_URI
+    );
+  }
+
+  isDevicePairingAsSupplicant() {
+    return DEVICE_PAIRING_SUPPLICANT_PATHNAME_REGEXP.test(this.pathname);
```

Comment:
unhelpful observation: Since this is a fairly basic RegExp, I think you could also use the `String#startsWith()` API:

```diff
const DEVICE_PAIRING_SUPPLICANT_PATHNAME = "/pair/supp/";
…
-return DEVICE_PAIRING_SUPPLICANT_PATHNAME_REGEXP.test(this.pathname);
+return this.pathname.startsWith(DEVICE_PAIRING_SUPPLICANT_PATHNAME);
```

Reply:
I'll take a look at this in that refactor follow up I keep mentioning in my self-review heh. I'm just using the relier flags as a base instead for now which has this same line.